modules/dnn/src/onnx/onnx_importer.cpp

```diff
@@ -1753,8 +1753,16 @@ void ONNXImporter::parseLSTM(LayerParams& layerParams, const opencv_onnx::NodePr
     CV_Assert(shapeIt != outShapes.end());
     const MatShape x_shape = shapeIt->second;
 
-    const int seq_length = x_shape[0];
-    const int batch_size = x_shape[1];
+    //if layout is 1, change batch and sequence dims
+    const int layout = layerParams.get<int>("layout", false);
```

Comment:
default value should be 0. See https://github.com/onnx/onnx/blob/906884ab594d5631cb4d31cf0d5aaeb78ed04f80/docs/Operators.md?plain=1#L19104

Reply:
Done!